app/assets/javascripts/live_search.js

```diff
@@ -134,50 +160,63 @@
-          this.trackingInit()
-          this.setRelevantResultCustomDimension()
-          this.trackPageView()
-        }.bind(this)
-      )
+      this.updateResults()
+    }
+  }
+
+  LiveSearch.prototype.serializeState = function (state) {
+    var params = []
+    if (Array.isArray(state)) {
+      for (var i = 0; i < state.length; i++) {
+        params.push(state[i].name + '=' + state[i].value)
+      }
+    } else {
+      for (var key in state) {
+        if (Object.prototype.hasOwnProperty.call(state, key)) {
+          params.push(encodeURIComponent(key) + '=' + encodeURIComponent(state[key]))
+        }
+      }
     }
+    return params.join('&')
   }
 
   LiveSearch.prototype.setRelevantResultCustomDimension = function setRelevantResultCustomDimension () {
-    var $mostRelevantDocumentLink = $('.js-finder-results').find('.gem-c-document-list__item--highlight')
-    var dimensionValue = $mostRelevantDocumentLink.length ? 'yes' : 'no'
+    var $mostRelevantDocumentLink = this.$form.querySelector('.gem-c-document-list__item--highlight')
+    var dimensionValue = $mostRelevantDocumentLink ? 'yes' : 'no'
     GOVUK.SearchAnalytics.setDimension(83, dimensionValue)
   }
 
   LiveSearch.prototype.trackingInit = function trackingInit () {
-    GOVUK.modules.start($(this.$resultsWrapper))
+    GOVUK.modules.start($(this.$resultsWrapper)) // this still needs to pass a jQuery object until we update the modules code
     this.indexTrackingData()
     this.startEnhancedEcommerceTracking()
   }
 
   LiveSearch.prototype.trackPageView = function trackPageView () {
-    var newPath = window.location.pathname + '?' + $.param(this.state)
+    var newPath = window.location.pathname + '?' + this.serializeState(this.state)
     GOVUK.SearchAnalytics.trackPageview(newPath)
     GOVUK.SearchAnalytics.trackPageview(newPath, document.title, { trackerName: 'govuk' })
   }
 
   LiveSearch.prototype.trackSpellingSuggestionsImpressions = function trackSpellingSuggestionsImpressions ($suggestions) {
-    var $spellingSuggestionMetaTag = $("meta[name='govuk:spelling-suggestion']")
-    // currently there's ever only one suggestion
-    var spellingSuggestionAvailable = this.$suggestionsBlock.find('a').length > 0
-    var suggestion = spellingSuggestionAvailable ? this.$suggestionsBlock.find('a').data('track-options').dimension81 : ''
-    $spellingSuggestionMetaTag.attr('content', suggestion)
+    var $spellingSuggestionMetaTag = document.querySelector("meta[name='govuk:spelling-suggestion']")
+    if (this.$suggestionsBlock && $spellingSuggestionMetaTag) {
+      var spellingSuggestionAvailable = this.$suggestionsBlock.querySelectorAll('a').length > 0
+      // currently there's ever only one suggestion
+      var link = this.$suggestionsBlock.querySelector('a')
+      var suggestion = ''
+      if (link) {
+        link = JSON.parse(link.getAttribute('data-track-options'))
+        suggestion = spellingSuggestionAvailable ? link.dimension81 : ''
```

Comment:
I don't think we need this condition since to have a link a spelling suggestion should be available (unless I'm misunderstanding something).

Reply:
Good point, have rewritten this section.